client/src/utils_links.js

```diff
@@ -24,7 +24,22 @@ const ENTITY_TYPE_TO_RELATED_OBJECT_TYPE = flip(
 )
 
 export function getEntityInfoFromUrl(url) {
-  const urnMatch = url.match(/^urn:anet:([^:]*):(.*)$/)
+  const urlObject = new URL(url, window.location.href)
+  if (urlObject.protocol === "urn:") {
+    return getEntityFromUrlPattern(
+      url,
+      urlObject.pathname,
+      /^anet:([^:]*):(.*)$/
+    )
+  }
+  if (urlObject.hostname === window.location.hostname) {
+    return getEntityFromUrlPattern(url, urlObject.pathname, /^\/([^/]*)\/(.*)$/)
+  }
+  return { type: EXTERNAL_LINK, url }
+}
+
+function getEntityFromUrlPattern(url, urlPath, pattern) {
+  const urnMatch = urlPath.match(pattern)
```

Comment:
Probably rename `pattern` to `entityPattern` and `urnMatch` to `entityMatch` (or something like that) to make the purpose a bit clearer.